### ISSUE TITLE
specifying Rails version in migration files

### ISSUE THREAD

**Issue by ekremkaraca** [norman/friendly_id#797]
I gave a try friendly_id in Rails 5.1 beta. During migration, I got error like this:

```shell
rails aborted!
StandardError: An error has occurred, this and all later migrations canceled:

Directly inheriting from ActiveRecord::Migration is not supported. Please specify the Rails release the migration was written for:

  class CreateFriendlyIdSlugs < ActiveRecord::Migration[4.2]
/home/ekrem/Projects/ekomerz/db/migrate/20170308164827_create_friendly_id_slugs.rb:1:in `<top (required)>'
/home/ekrem/Projects/ekomerz/bin/rails:9:in `require'
/home/ekrem/Projects/ekomerz/bin/rails:9:in `<top (required)>'
/home/ekrem/Projects/ekomerz/bin/spring:15:in `<top (required)>'
bin/rails:3:in `load'
bin/rails:3:in `<main>'
StandardError: Directly inheriting from ActiveRecord::Migration is not supported. Please specify the Rails release the migration was written for:

  class CreateFriendlyIdSlugs < ActiveRecord::Migration[4.2]
/home/ekrem/Projects/ekomerz/db/migrate/20170308164827_create_friendly_id_slugs.rb:1:in `<top (required)>'
/home/ekrem/Projects/ekomerz/bin/rails:9:in `require'
/home/ekrem/Projects/ekomerz/bin/rails:9:in `<top (required)>'
/home/ekrem/Projects/ekomerz/bin/spring:15:in `<top (required)>'
bin/rails:3:in `load'
bin/rails:3:in `<main>'
Tasks: TOP => db:migrate
(See full trace by running task with --trace)
```

In migration file, specifying Rails version on first line of the migration file like below solves the problem:
```ruby
class CreateFriendlyIdSlugs < ActiveRecord::Migration[5.1]
```

**Comment by scottbaggett** [comment]
Just ran into this same issue. +1

**Comment by nmsiriban** [comment]
I ran into the same problem as well. +1

**Comment by aconstandinou** [comment]
@nmsiriban I had the same issue as the OP, and your fix worked for me - thank you

**Comment by aaron-price** [comment]
Solved it for me too. Thanks!

**Comment by giedriusr** [comment]
Seems like the gem has been abandoned, no?

**Comment by agrohmann** [comment]
+1

**Comment by wafindotca** [comment]
work like charm , thank you

**Comment by LuisMenjivar** [comment]
I encountered this error when I upgraded to rails 5.1 then reset my database and then tried to migrate. 
specify the Rails release the migration was written for like it was shown solved my issue. Should read the error messages carefully before panicking and googleing everything. Thanks!

**Comment by cristivania** [comment]
Solve it! Thank You!

**Comment by ninja247** [comment]
@nmsiriban thank you!!!!

great to see another devcamp guy on here!!!

**Comment by avishekdas2018** [comment]
Oh solve it!!! thanks man!

**Comment by desenar** [comment]
Thank you very much!!!!

**Comment by etagwerker** [comment]
FWIW, this seems to be fixed in the `master` branch: https://github.com/norman/friendly_id/blob/a29e7d7cc280d7d46c181f58fabd7cc9be67eefe/lib/friendly_id/migration.rb#L2-L6

**Comment by phellykz** [comment]
muito obrigado!

**Comment by tvanderpol** [comment]
Yup this issue is resolved in the current version of the gem (v5.2.4).

**Comment by azeroual2017** [comment]
Thank you!

**Comment by LeVanVuong93** [comment]
> FWIW, this seems to be fixed in the `master` branch:
> 
> [friendly_id/lib/friendly_id/migration.rb](https://github.com/norman/friendly_id/blob/a29e7d7cc280d7d46c181f58fabd7cc9be67eefe/lib/friendly_id/migration.rb#L2-L6)
> 
> Lines 2 to 6 in [a29e7d7](/norman/friendly_id/commit/a29e7d7cc280d7d46c181f58fabd7cc9be67eefe)
> 
>  if ActiveRecord::VERSION::MAJOR >= 5 
>    ActiveRecord::Migration[4.2] 
>  else 
>    ActiveRecord::Migration 
>  end

Hi etagwerker,
What is the folder need modify to fix that error. Could you tell me, Pls

**Comment by parndt** [comment]
@LeVanVuong93 https://github.com/norman/friendly_id/issues/797#issuecomment-402357003

>Yup this issue is resolved in the current version of the gem (v5.2.4).
